### ISSUE TITLE
allow exception for SC1007 Remove space after = if trying to assign a value

### ISSUE THREAD

**Issue by bittorf** [koalaman/shellcheck#493]
SC1007 is there to avoid mistakes.
here are examples, which are clearly no mistakes

```sh
#!/bin/sh
export LANADR=
VAR1=;VAR2=;VAR3=

x() {
  local var=
  ...
}
```

can we please stop to warn here? writing VAR="" seems to be overengineered.

**Comment by koalaman** [comment]
Which version are you using? Copy-pasting this into shellcheck.net (runnint latest git commit) does not produce any SC1007s.

**Comment by bittorf** [comment]
sorry, ofcourse i'am using also the latest git and the testcase is this;
```shell
#!/bin/sh
A= B=1
echo $A
echo $B
```

**Comment by koalaman** [comment]
I'd agree that no warning is needed for `A=    #comment`. This is fixed in d01b59a.

This still warns for `local a= b=1`, which in any other langnuage would assign a=1 but is empty here.
